src/App.js

```diff
@@ -1,61 +1,87 @@
 import React, { Component } from 'react'
 import logo from './logo.png';
 import './App.css';
-// import movieData from "./data";
-// import Movie from "./Movie";
+import Movie from "./Movie";
 import MovieProfile from "./MovieProfile"
 
-const defaultMovie = {
-  id: 1,
-  title: "Fake Movie Title",
-  poster_path: "https://image.tmdb.org/t/p/original//7G2VvG1lU8q758uOqU6z2Ds0qpA.jpg",
-  backdrop_path: "https://image.tmdb.org/t/p/original//oazPqs1z78LcIOFslbKtJLGlueo.jpg",
-  release_date: "2019-12-04",
-  overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!",
-  average_rating: 6,
-  genres: [{id: 18, name: "Drama"}, {id: 42, name: "Comedy"}],
-  budget: 63000000,
-  revenue: 100853753,
-  runtime: 139,
-  tagline: "It's a movie!"
-}
-
 class App extends Component {
+  _isLoaded = false
+
   constructor() {
     super();
+
     this.state = {
-      defaultMovie
+      allMovies: [],
+      currentMovie: this.getMovie(659986), // change this once onClick is fixed, to {}
+      error: ''
     }
   }
 
+  componentDidMount() {
+    this._isLoaded = true;
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies`)
+      .then(response => response.json())
+      .then(allMovies => this.setState({ allMovies: allMovies.movies }))
+      .catch(error => this.setState({ error: error.message }))
+  }
+
+  getMovie(id) {
+    this._isLoaded = true; // this isn't working correctly for getMovie, maybe because I'm calling it in the constructor
```

Comment:
Also, instead of having a boolean for loading, it would probably be better to just check the other data in state to determine if your page is "loading" or not.

Reply:
Ok, I'll refactor that. thanks!!